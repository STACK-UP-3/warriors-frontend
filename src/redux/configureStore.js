import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
