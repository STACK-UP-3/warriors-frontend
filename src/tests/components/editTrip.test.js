import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import EditTrip from '../../components/editTripRequest';
import configureStore from '../../redux/configureStore';

describe('*************** Testing the EditTrip  component ***************',()=>{
    const store = configureStore();

    it('Should render the EditTrip page correctly', ()=>{
        const wrapper = mount( 
            <Provider store={ store }> 
                <EditTrip  match={{params:{id:1}}}/> 
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
