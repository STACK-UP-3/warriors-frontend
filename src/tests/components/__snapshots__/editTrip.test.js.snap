// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*************** Testing the EditTrip  component *************** Should render the EditTrip page correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <_default
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  >
    <div>
      <h5>
        Lets EDIT TRIP REQUEST : 
        1
      </h5>
    </div>
  </_default>
</Provider>
`;
